packages/orm/src/relations/ManyToManyCollection.ts

```diff
@@ -228,23 +228,28 @@ export class ManyToManyCollection<T extends Entity, U extends Entity>
       const m2m = other[this.otherFieldName] as any as ManyToManyCollection<U, T>;
       m2m.remove(this.entity);
     });
-    this.loaded = [];
+    this.#loaded = [];
   }
 
   private maybeApplyAddedAndRemovedBeforeLoaded(): void {
-    if (this.loaded) {
-      // this.loaded.unshift(...this.addedBeforeLoaded);
-      // this.addedBeforeLoaded = [];
-      this.removedBeforeLoaded?.forEach((other) => {
-        remove(this.loaded!, other);
-        getEmInternalApi(this.entity.em).joinRows(this).addRemove(this, this.entity, other);
+    if (this.#loaded) {
+      this.#addedBeforeLoaded?.forEach((e) => {
+        if (!this.#loaded?.includes(e)) {
```

Comment:
Fwiw I don't love the "for each added, scan loaded.includes", it would technically be an N^2 on large collections...